share/functions/fish_git_prompt.fish

```diff
@@ -270,15 +271,21 @@ function fish_git_prompt --description "Prompt function for Git"
     contains untrackedfiles $__fish_git_prompt_status_order
     or set untracked false
 
+    contains -- "$__fish_git_prompt_showstashstate" yes true 1
+    and set stash true
+    contains stashstate $__fish_git_prompt_status_order
+    or set stash false
+
     if test true = $inside_worktree
         # Use informative status if it has been enabled locally, or it has been
-        # enabled globally (via the fish variable) and dirty or untracked are not false.
+        # enabled globally (via the fish variable) and dirty, untracked, and stash are not false.
+        # (supports #9572)
         #
         # This is to allow overrides for the repository.
         if test "$informative" = true
             or begin
                 contains -- "$__fish_git_prompt_show_informative_status" yes true 1
-                and test "$dirty" != false
+                and test "$dirty" != false -o "$untracked" != false -o "$stash" != false
```

Comment:
This means that whenever you ask for stash state it will switch to informative mode, which is *much* slower.

It's probably worth preserving stash state that doesn't also incorporate calling `git status` to count everything.